### PR TITLE
feat: add bigquery mod op

### PR DESCRIPTION
**Pull request by eakmanrq**
Screenshot 2024-03-17 at 8 08 38 PM
Screenshot 2024-03-17 at 8 09 01 PM

### CODE REVIEW

File: tests/dialects/test_bigquery.py [hunk 1]
```diff
@@ -1090,6 +1090,13 @@ def test_bigquery(self):
             },
             pretty=True,
         )
+        self.validate_all(
+            "SELECT x % 10",
```
Comment (tobymao) [inline tests/dialects/test_bigquery.py:1094]: this isn't valid bigquery, it should be mod(x, 10) right?

File: sqlglot/dialects/bigquery.py [hunk 2]
```diff
@@ -898,6 +898,9 @@ def intersect_op(self, expression: exp.Intersect) -> str:
                 self.unsupported("INTERSECT without DISTINCT is not supported in BigQuery")
             return f"INTERSECT{' DISTINCT' if expression.args.get('distinct') else ' ALL'}"
 
+        def mod_sql(self, expression: exp.Mod) -> str:
```
Comment (tobymao) [inline sqlglot/dialects/bigquery.py:901]: for simple things like this, you can just put it in transforms, i think you could do

exp.Mod: rename_func("MOD")

File: tests/dialects/test_bigquery.py [hunk 3]
```diff
@@ -1090,6 +1090,13 @@ def test_bigquery(self):
...
+            write={
```
Comment (tobymao) [inline tests/dialects/test_bigquery.py:1095]: you could do read={
  postgres: "select x % 10" to test this
}

Comment (eakmanrq) [reply tests/dialects/test_bigquery.py:1096]: I get `SELECT MOD(x, 10)` in postgres now but that is valid: https://www.postgresqltutorial.com/postgresql-math-functions/postgresql-mod/

Comment (tobymao) [reply tests/dialects/test_bigquery.py:1096]: it's being parsed as anonymous, you need to add a parser

FUNCTIONs = { "MOD": ...}

Comment (eakmanrq) [reply tests/dialects/test_bigquery.py:1096]: Updated: https://github.com/tobymao/sqlglot/pull/3157/commits/8c161bb7547e458f0c94898f78dbb9fdbf3470b4

**Review by tobymao — APPROVED** [review]

File: tests/dialects/test_duckdb.py [hunk 4]
```diff
@@ -725,6 +725,7 @@ def test_array_index(self):
             )
 
     def test_time(self):
+        self.maxDiff = None
```
Comment (tobymao) [inline tests/dialects/test_duckdb.py:728]: delete this

Comment (eakmanrq) [reply tests/dialects/test_duckdb.py:728]: oops thanks

**Pull request merged**